packages/effect/src/Array.ts

```diff
@@ -145,6 +145,24 @@ export const replicate: {
 export const fromIterable = <A>(collection: Iterable<A>): Array<A> =>
   Array.isArray(collection) ? collection : Array.from(collection)
 
+/**
+ * Creates a new `Array` from a value that might not be an iterable.
+ *
+ * @example
+ * import { Array } from "effect"
+ *
+ * assert.deepStrictEqual(Array.ensure("a"), ["a"])
+ * assert.deepStrictEqual(Array.ensure(["a"]), ["a"])
+ * assert.deepStrictEqual(Array.ensure(["a", "b", "c"]), ["a", "b", "c"])
+ *
+ * const set = new Set([1, 2, 3])
+ * assert.deepStrictEqual(Array.ensure(set), [1, 2, 3])
+ *
+ * @category constructors
+ * @since 3.3.0
+ */
+export const ensure = <A>(self: ReadonlyArray<A> | A): Array<A> => Array.isArray(self) ? self : of(self)
```

Comment:
```suggestion
export const ensure = <A>(self: ReadonlyArray<A> | A): Array<A> => Array.isArray(self) ? self.slice() : of(self)
```

Reply:
`fromIterable` does not clone the array, and if it should then we should remove it since it would `===` the vanilla `Array.from` lol
```ts
export const fromIterable: <A>(self: Iterable<A>): Array<A> = Array.from // 🤨
export const ensure = <A>(self: ReadonlyArray<A> | A): Array<A> => Array.from([self].flat(1)) // if we allow cloning
```

Reply:
Array.from performs a shallow copy:

<img width="488" alt="image" src="https://github.com/Effect-TS/effect/assets/40680/2971e60b-f6ae-4822-88b6-95576ba62031">

Reply:
Is this an anecdote, or an argument for keeping `fromIterable` because it only clones non-`Array`s?

Reply:
It is there because if you `import { Array } from "effect"`, accessing `Array.from` becomes annoying.

Reply:
So in conclusion we need to have `fromIterable` do a `slice()` and so does `ensure`